### PR TITLE
[Telescope#2755] Adding more tests for createError

### PR DESCRIPTION
**Pull request by rclee91**
Requested from [#2755](https://github.com/Seneca-CDOT/telescope/issues/2755)

**What I did**
Added more tests to `createError`

**What was found?**
It's ElasticSearch's fault. They have errors that are structured differently than regular `http-errors` causing the failure.

**How to test**
`pnpm test`

### CODE REVIEW

**Comment by rclee91** [comment]
Added more tests. Shows how we can indirectly createError from an ES Error

**Comment by AmasiaNalbandian** [comment]
holy smokes Roxanne!!! I'm running it now. :)

File: test.js [hunk 1]
```diff
@@ -893,12 +893,75 @@ describe('hash', () => {
...
+
+  test('should fail when directly creating Error from ElasticSearch error object', () => {
+    // { errors } is imported from ElasticSearch
+    const elasticError = new errors.ResponseError({
+      body: { error: 'testing ElasticSearch Error' },
+      statusCode: 404,
+    });
+
+    try {
+      createError(503, elasticError);
```
Comment (AmasiaNalbandian) [inline test.js:929]: were bringing back a 503 because elastic returns 404 not available correct?

**Review by AmasiaNalbandian — APPROVED** [review]

File: test.js (hunk 1 above)
Comment (rclee91) [reply test.js:929]: I'm just replicating what's being done in the `query.js` that's causing the error
```javascript
// query.js
router.get('/', validateQuery, async (req, res, next) => {
  try {
    const { text, filter, page, perPage } = req.query;
    res.send(await search(text, filter, page, perPage));
  } catch (error) {
    console.log({ error }, 'query error');
    next(createError(503, error));   // <-- crash is here
  }
});
```
If you look below in the indirect test, there are different ways we can send in the "correct" status code. 
The thing is though, ES has different status code names for their errors than the regular http errors. Even if we send the ES status code in for `createError`, it'll just name it depending on the http error and not the ES error.

**Review by humphd — APPROVED** [review]
Question: in a follow-up, since we expose `createError` and `Elastic` from Satellite, should we do something internally to have this work as expected by our users?  That is, should we add to `createError()` an extra case to deal with extracting the right bits from an Elastic error automatically?

**Comment by rclee91** [comment]
Yes, I think we probably could.

I think right now in Satellite there's no module for `createError`, it's just exporting `createError` from `http-errors`
[Link to line](https://github.com/Seneca-CDOT/satellite/blob/4b4fd2224e2b41181eff843e8751bdafe008f9b1/src/index.js#L7)
```js
module.exports.createError = require('http-errors');
```
We could probably create its own module to modify it the way we want to. 

Based on the [documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/breaking-changes.html), all of ES errors should be called `ResponseError`
```text
Errors: there is no longer a custom error class for every HTTP status code (such as BadRequest or NotFound). 
There is instead a single ResponseError. Every error class has been renamed, and now each is suffixed with Error at the end. 
```
I have been looking at the[ list of http-errors](https://www.npmjs.com/package/http-errors). It doesn't seem like there's one called `ResponseError`, so we might be able to filter errors based off of its name, and then extract the ES error. 

The question perhaps is just how much detail, or which object, we want to send in to `createError`

**Comment by rclee91** [comment]
I'm not sure if I have it down right. But since `createError` is an imported function, does that mean it is a `const` and if we want to export our own versions, maybe we'd have to rename our exported function?

**Pull request merged**